src/entities/protocols/unwrapWETH.ts

```diff
@@ -26,8 +23,10 @@ export class UnwrapWETH implements Command {
   }
 
   encode(planner: RoutePlanner, _: TradeConfig): void {
-    if (!!this.permit2Data) encodePermit(planner, this.permit2Data)
-    planner.addCommand(CommandType.PERMIT2_TRANSFER_FROM, [this.wethAddress, this.routerAddress, this.amount])
+    encodeHandleInputTokens(planner, undefined, this.permit2Data, {
```

Comment:
I personally tend to be pretty averse to functions that can have random `undefined`'s sprinkled in its function parameters.
Personal preference, but I think we also tend to do this in other sdk's maybe there could be options when not everything is required:

```
type InputTokenOptions {
   approval?: ApproveProtocol,
   permit2Data? Permit2Permit,
   permit2TransferFrom?: Permit2TransferFromOptions
}

encodeHandleInputTokens(
  planner, 
  { 
     permit2Data: this.permit2Data,
     permit2TransferFrom: {
         token: this.wethAddress,
         amount: this.amount.toString()
  }

)
```

Reply:
ah i prefer this too fwiw